Peephole Optimize out conv(x).dim(), which prevents BN fusion

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50241 Re-enable no-op peepholes
* #50222 Add Post Freezing Optimizations, turn on by default in torch.jit.freeze
* **#50221 Peephole Optimize out conv(x).dim(), which prevents BN fusion**
* #50220 [JIT] Factor out peephole to own test file
* #50075 [JIT] Add Frozen Conv-> Add/Sub/Mul/Div fusion
* #50074 [JIT] Frozen Graph Conv-BN fusion

Differential Revision: [D25856266](https://our.internmc.facebook.com/intern/diff/D25856266)

Commit:
Peephole Optimize out conv(x).dim(), which prevents BN fusion

[ghstack-poisoned]

Commit:
Update on "Peephole Optimize out conv(x).dim(), which prevents BN fusion"




[ghstack-poisoned]

Commit:
Update on "Peephole Optimize out conv(x).dim(), which prevents BN fusion"




[ghstack-poisoned]

Commit:
Update on "Peephole Optimize out conv(x).dim(), which prevents BN fusion"




[ghstack-poisoned]

Krovatkin commented on torch/csrc/jit/passes/peephole_alias_sensitive.cpp:
```diff
@@ -0,0 +1,79 @@
+        } else {
+          for (const Use& dim_use : dim_uses) {
+            if (aliasDb_->moveAfterTopologicallyValid(node, dim_use.user)) {
+              replaceWithIValue(dim_use.user->output(), output_size);
```
Just to make sure I understand this part:
if we have multiple `dim`s that we can move `node` to (without any `resizes_` in between). We would move `conv` multiple times?

eellison replied:
We're repeatedly trying to move `aten::dim` to the conv node. Not the other way around.

Krovatkin replied:
ah, misread the api comment, thanks!

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "Peephole Optimize out conv(x).dim(), which prevents BN fusion"


Differential Revision: [D25856266](https://our.internmc.facebook.com/intern/diff/D25856266)

[ghstack-poisoned]
